src/GraphQL.AspNetCore.GraphQL/GraphQLMiddleware.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using GraphQL.Http;
+using GraphQL.Types;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace GraphQL.AspNetCore.GraphQL {
+    /// <summary>
+    ///     Provides middleware for hosting GraphQL.
+    /// </summary>
+    public sealed class GraphQLMiddleware {
+        private readonly string graphqlPath;
+        private readonly RequestDelegate next;
+        private readonly ISchema schema;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="GraphQLMiddleware" /> class.
+        /// </summary>
+        /// <param name="next">
+        ///     The next request delegate.
+        /// </param>
+        /// <param name="options">
+        ///     The GraphQL options.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        ///     Throws <see cref="ArgumentNullException" /> if <paramref name="next" /> or <paramref name="options" /> is null.
+        /// </exception>
+        public GraphQLMiddleware(RequestDelegate next, IOptions<GraphQLOptions> options) {
+            if(next == null) {
+                throw new ArgumentNullException(nameof(next));
+            }
+            if(options == null) {
+                throw new ArgumentNullException(nameof(options));
+            }
+            if(options.Value?.Schema == null) {
+                throw new ArgumentException("Schema is null");
+            }
+
+            this.next = next;
+            var optionsValue = options.Value;
+            graphqlPath = string.IsNullOrEmpty(optionsValue?.GraphQLPath) ? GraphQLOptions.DefaultGraphQLPath : optionsValue.GraphQLPath;
+            schema = optionsValue?.Schema;
+        }
+
+        /// <summary>
+        ///     Invokes the middleware with the specified context.
+        /// </summary>
+        /// <param name="context">
+        ///     The context.
+        /// </param>
+        /// <returns>
+        ///     A <see cref="Task" /> representing the middleware invocation.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///     Throws <see cref="ArgumentNullException" /> if <paramref name="context" />.
+        /// </exception>
+        public async Task Invoke(HttpContext context) {
+            if(context == null) {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if(ShouldRespondToRequest(context.Request)) {
+                var executionResult = await ExecuteAsync(context.Request).ConfigureAwait(true);
+                await WriteResponseAsync(context.Response, executionResult).ConfigureAwait(true);
```

Comment:
You shouldn't call next if you have responded here.  Whats with the .ConfigureAwait(true) all over the place? There shouldn't be any need to keep the original context.

This and the errors check previously discussed are the only two bugs I have I found with this PR. GraphQL.AspNetCore.GraphiQL.index.html seems to be missing too but I didn't test the explorer middleware.